__tests__/app.test.js

```diff
@@ -198,3 +194,65 @@ describe("/api/reviews/:review_id/comments", () => {
       });
   });
 });
+describe("POST /api/reviews/:review_id/comments", () => {
+  it("responds with the posted comment object", () => {
+    const comment = { username: "mallionaire", body: "test comment" };
+    return request(app)
+      .post("/api/reviews/1/comments")
+      .send(comment)
+      .expect(201)
+      .then((response) => {
+        const expectedComment = {
+          comment_id: expect.any(Number),
+          author: comment.username,
+          body: comment.body,
+          created_at: expect.any(String),
+          review_id: 1,
+          votes: expect.any(Number),
+        };
+        expect(response.body.comment).toMatchObject(expectedComment);
+      });
+  });
+  it("responds with 400 Bad Request when given an invalid review ID", () => {
+    const comment = { username: "mallionaire", body: "test comment" };
+    return request(app)
+      .post("/api/reviews/not_an_id/comments")
+      .send(comment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Bad Request");
+      });
+  });
+  it("responds with 404 Not Found when given a non-existent username", () => {
+    const comment = { username: "nonexistentuser", body: "test comment" };
+    return request(app)
+      .post("/api/reviews/1/comments")
+      .send(comment)
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("User not found");
+      });
+  });
+  it("201: it should ignore unnecessary properties", () => {
+    const comment = {
+      username: "mallionaire",
+      body: "test comment",
+      faveAnimal: "snake",
+    };
+    return request(app)
+      .post("/api/reviews/1/comments")
+      .send(comment)
+      .expect(201)
+      .then((response) => {
+        const expectedComment = {
+          comment_id: expect.any(Number),
+          author: comment.username,
+          body: comment.body,
+          created_at: expect.any(String),
+          review_id: 1,
+          votes: expect.any(Number),
+        };
```

Comment:
Just need a 400 non exist ID 9999999 and 400 missing required fields e.g. no username or body properties

Reply:
I thought non exist ID 999999 would be a 404 because it doesnt exist yet but still a valid path. will just return empty results?